src/main/java/fr/insalyon/citi/golo/compiler/JavaBytecodeGenerationGoloIrVisitor.java

```diff
@@ -751,22 +751,27 @@ public void visitTryCatchFinally(TryCatchFinally tryCatchFinally) {
   @Override
   public void visitClosureReference(ClosureReference closureReference) {
     GoloFunction target = closureReference.getTarget();
-    boolean isVarArgs = target.isVarargs();
-    int arity = (isVarArgs) ? target.getArity() - 1 : target.getArity();
+    final boolean isVarArgs = target.isVarargs();
+    final int arity = (isVarArgs) ? target.getArity() - 1 : target.getArity();
+    final int syntheticCount = closureReference.getTarget().getSyntheticParameterCount();
     methodVisitor.visitInvokeDynamicInsn(
         target.getName(),
-        methodType(MethodHandle.class).toMethodDescriptorString(),
+        methodType(FunctionReference.class).toMethodDescriptorString(),
         CLOSUREREF_HANDLE,
         klass,
         (Integer) arity,
         (Boolean) isVarArgs);
-    final int syntheticCount = closureReference.getTarget().getSyntheticParameterCount();
     if (syntheticCount > 0) {
-      ReferenceTable table = context.referenceTableStack.peek();
+      methodVisitor.visitMethodInsn(
+          INVOKEVIRTUAL,
+          "gololang/FunctionReference",
+          "handle",
+          "()Ljava/lang/invoke/MethodHandle;", false);
       String[] refs = closureReference.getCapturedReferenceNames().toArray(new String[syntheticCount]);
       loadInteger(methodVisitor, 0);
       loadInteger(methodVisitor, syntheticCount);
       methodVisitor.visitTypeInsn(ANEWARRAY, "java/lang/Object");
+      ReferenceTable table = context.referenceTableStack.peek();
```

Comment:
why do we need to add a `peek`. I'm just trying to improve my bytecode skills not challenging the impl.